astrovascpy/bloodflow.py

```diff
@@ -653,6 +649,15 @@ def _solve_linear(laplacian, input_flow):
         scipy.sparse.csc_matrix: frequency dependent laplacian matrix
     """
 
+    WARNING_MSG = textwrap.dedent(
+        """ Number of nodes = %(n_nodes)s.
+    The program can be slow and the result not very accurate.
+    It is recommended to use PETSc for a graph with more than 2e6 nodes and
+    SciPy for a graph with less than 2e6 nodes.
+    The default solver can be selected in setup.sh or by setting the environment
+    variable BACKEND_SOLVER_BFS to 'scipy' or 'petsc'. """
+    )
```

Comment:
`dedent` removes the number of leading whitespace of the first line in every line of the text. In your case, `dedent` will only remove one space character, the result string will be:

```
Number of nodes = %(n_nodes)s.
   The program can be slow and the result not very accurate.
   It is recommended to use PETSc for a graph with more than 2e6 nodes and
   SciPy for a graph with less than 2e6 nodes.
   The default solver can be selected in setup.sh or by setting the environment
   variable BACKEND_SOLVER_BFS to 'scipy' or 'petsc'.
```

```suggestion
    WARNING_MSG = textwrap.dedent("""\
    Number of nodes = %(n_nodes)s.
    The program can be slow and the result not very accurate.
    It is recommended to use PETSc for a graph with more than 2e6 nodes and
    SciPy for a graph with less than 2e6 nodes.
    The default solver can be selected in setup.sh or by setting the environment
    variable BACKEND_SOLVER_BFS to 'scipy' or 'petsc'. """
    )
```
With this suggestion, the result of the `dedent` operation is:
```
Number of nodes = %(n_nodes)s.
The program can be slow and the result not very accurate.
It is recommended to use PETSc for a graph with more than 2e6 nodes and
SciPy for a graph with less than 2e6 nodes.
The default solver can be selected in setup.sh or by setting the environment
variable BACKEND_SOLVER_BFS to 'scipy' or 'petsc'.
```

Reply:
Ah ok, thanks!